addons/payment_stripe/static/src/js/payment_form.js

```diff
@@ -75,9 +76,67 @@ PaymentForm.include({
                     window.location.reload();
                 }
             } else {
-                $checkedRadio.val(result.id);
-                self.el.submit();
+                var json_url = self.$el.data('json-action');
+                formData.pm_id = result.id;
+                return self._rpc({
+                    route: json_url,
+                    params: formData
+                });
+            }
+        }).then(function(result) {
+            var pi_secret = result.stripe_payment_intent_secret;
+            tx_reference = result.reference;
+            return stripe.handleCardPayment(pi_secret);
+        }).then(function(result) {
+            if (result.error) {
+                return Promise.reject({"message": {"data": { "message": result.error.message}}});
+            }
+            console.log(result);
+            return self._rpc({
+                route: '/payment/stripe/s2s/process_payment_intent',
+                params: _.extend({}, result.paymentIntent, {reference: tx_reference}),
+            })
+        }).then(function(result) {
+            window.location = '/payment/process';
+        }).guardedCatch(function (error) {
+            // if the rpc fails, pretty obvious
+            self.enableButton(button);
+            self.displayError(
+                _t('Unable to save card'),
+                _t("We are not able to add your payment method at the moment. ") +
+                    self._parseError(error)
+            );
+        });
+    },
+    _chargeStripeToken: function (ev, $checkedRadio) {
+        var self = this;
```

Comment:
Is it possible to create a separate method for creating charge as most of the code repeates in _chargeStripeToken and _createStripeToken??

Reply:
good point i'll factorize the common code in a private method

Reply:
done!